src/nix/formatter.cc

```diff
@@ -0,0 +1,168 @@
+        auto evalStore = getEvalStore();
+
+        auto installable_ = parseInstallable(store, ".");
+        auto & installable = InstallableValue::require(*installable_);
+        auto app = installable.toApp(*evalState).resolve(evalStore, store);
+
+        Strings programArgs{app.program};
+
+        // Propagate arguments from the CLI
+        for (auto & i : args) {
+            programArgs.push_back(i);
+        }
+
+        // Release our references to eval caches to ensure they are persisted to disk, because
+        // we are about to exec out of this process without running C++ destructors.
+        evalState->evalCaches.clear();
+
+        execProgramInStore(store, UseLookupPath::DontUse, app.program, programArgs);
+    };
+};
+
+static auto rFormatterRun = registerCommand2<CmdFormatterRun>({"formatter", "run"});
+
+struct CmdFormatterBuild : MixFormatter
+{
+    Path outLink = "result";
+
+    CmdFormatterBuild()
+    {
+        addFlag({
+            .longName = "out-link",
+            .shortName = 'o',
+            .description = "Use *path* as prefix for the symlink to the build result. It defaults to `result`.",
+            .labels = {"path"},
+            .handler = {&outLink},
+            .completer = completePath,
+        });
+
+        addFlag({
+            .longName = "no-link",
+            .description = "Do not create symlink to the build results.",
+            .handler = {&outLink, Path("")},
+        });
+    }
+
+    std::string description() override
+    {
+        return "build the current flake's formatter";
+    }
+
+    std::string doc() override
+    {
+        return
+#include "formatter-build.md"
+            ;
+    }
+
+    Category category() override
+    {
+        return catSecondary;
+    }
+
+    void run(ref<Store> store) override
+    {
+        auto evalState = getEvalState();
+        auto evalStore = getEvalStore();
+
+        auto installable_ = parseInstallable(store, ".");
+        auto & installable = InstallableValue::require(*installable_);
+        auto unresolvedApp = installable.toApp(*evalState);
+        auto app = unresolvedApp.resolve(evalStore, store);
+
+        Installables installableContext;
+        for (auto & ctxElt : unresolvedApp.unresolved.context)
+            installableContext.push_back(make_ref<InstallableDerivedPath>(store, DerivedPath{ctxElt}));
+        auto buildables = Installable::build(evalStore, store, Realise::Outputs, installableContext);
+
+        if (outLink != "")
+            if (auto store2 = store.dynamic_pointer_cast<LocalFSStore>())
+                createOutLinks(outLink, toBuiltPaths(buildables), *store2);
+
+        logger->cout("%s", app.program);
+    };
+};
+
+static auto rFormatterBuild = registerCommand2<CmdFormatterBuild>({"formatter", "build"});
+
+struct CmdFmt : CmdFormatterRun
+{
+    void run(ref<Store> store) override
+    {
+        CmdFormatterRun::run(store);
+    }
+};
+
+static auto rFmt = registerCommand<CmdFmt>("fmt");
```

Comment:
This is a bit different from other command aliases but doesn't suffer from a missing man page, so let's keep it this way until we fix and harmonize that.